src/cli/cli_dataset.cpp

```diff
@@ -336,19 +344,23 @@ template <> otError Dataset::Process<Cmd("channel")>(Arg aArgs[])
  */
 template <> otError Dataset::Process<Cmd("channelmask")>(Arg aArgs[])
 {
-    otError error = OT_ERROR_NONE;
+    otError              error = OT_ERROR_NONE;
+    otOperationalDataset dataset;
 
     if (aArgs[0].IsEmpty())
     {
-        if (sDataset.mComponents.mIsChannelMaskPresent)
+        SuccessOrExit(error = otDatasetParseTlvs(&sDataset, &dataset));
+        if (dataset.mComponents.mIsChannelMaskPresent)
         {
-            OutputLine("0x%08lx", ToUlong(sDataset.mChannelMask));
+            OutputLine("0x%08lx", ToUlong(dataset.mChannelMask));
         }
     }
     else
     {
-        SuccessOrExit(error = aArgs[0].ParseAsUint32(sDataset.mChannelMask));
-        sDataset.mComponents.mIsChannelMaskPresent = true;
+        memset(&dataset, 0, sizeof(dataset));
+        SuccessOrExit(error = aArgs[0].ParseAsUint32(dataset.mChannelMask));
+        dataset.mComponents.mIsChannelMaskPresent = true;
+        SuccessOrExit(error = otDatasetUpdateTlvs(&dataset, &sDataset));
     }
```

Comment:
In the `else` part we need to clear all `mIs{}Present`?
Maybe we can add a helper method here `Clear(otOperationalDataset &aDataset)` and do `memset(0)`?

Reply:
I think you already did it 😄